### DISCUSSION TITLE
OpenWrt in a virtual machine

### DISCUSSION THREAD

**Discussion by tteck** [tteck/Proxmox#1435] [Announcements]
A script has been submitted by @JonTheNiceGuy for installing OpenWrt in a virtual machine, and we greatly appreciate their contribution. Please share your findings and observations here. 

```
bash -c "$(wget -qLO - https://github.com/tteck/Proxmox/raw/main/vm/openwrt.sh)"
```

**Comment by JonTheNiceGuy** [comment]
Just for context, this image is a little different to most images you'll see on Proxmox:

* This has two interfaces - a "WAN" interface (attached to vmbr0) and a "LAN" interface (provisionally attached to vmbr0 with a vlan tag of 999). The LAN interface runs a DHCP server, which will offer addresses in the documentation range; 192.0.2.0/24 (this needs to be changed following the installation)
* The WAN interface (which would usually be attached to the internet) exposes the management interface (so you can manage it from your "normal" network)... which is probably fine, except that there's no initial password on the root account (although that could be fixed).

The WAN management side of things could be configured as part of the script, as could the IP addressing, but I (selfishly) wanted to get my script uploaded *first* and then it can be tweaked afterwards :)

My work here was made so much easier by tteck's code so :heart: thanks! It was also made possible by [this random blog post](https://i12bretro.github.io/tutorials/0405.html) I found while looking for an OpenWRT-on-Proxmox instruction guide

**Comment by dmshimself** [comment]
I installed, configured and tested with a WAN and small dhcp served container and it all came up fine.  Great news.  I had in mind adding a USB wifi stick to this VM to provide a simple wifi AP.  Do you think this is likely to be possible?  So far I've added a couple of different sticks and installed a fair amount of extra kmod- modules from the software menu of openwrt and rebooted, but the extra device doesn't appear in network/interface/devices.  No worries if this isn't likely to work, but it would be an attractive option.

**Reply by JonTheNiceGuy** [reply]
I honestly don't know. If you've passed through the USB device through either the WebUI or via the [command line](https://pve.proxmox.com/wiki/USB_Devices_in_Virtual_Machines) then they probably should be available in the VM, but I've not tried, sorry!

**Comment by dmshimself** [comment]
Cheers - I'll have a good rummage around and report back.  Everything else looks great to me, ping, nslookup, reporting tools, all good

**Comment by dmshimself** [comment]
Yes I got it to go using standard openwrt methods.  I installed any kmod- modules needed for the USB card, rebooted, made sure I could see the usb device using lsusb which also needed to be installed.  I then ran wifi config, wifi and rebooted again and then I just had to enable the radio from the openwrt UI.  What someone does then is down to requirements.  In my case I deleted the eth1 setup and created a wireless AP

**Comment by rickicode** [comment]
How to resize the rootfs?
